### PR TITLE
tmaurer3/move_headers_to_one_include_folder

### PR DESCRIPTION
**Pull request by tmaurer3**

### CODE REVIEW

**Comment by tmaurer3** [comment]
@kmilos  Feel free to check if this works for you.

**Comment by kmilos** [comment]
Unfortunately it looks like CMake generators (Ninja at least) define `Lerc_EXPORTS` instead of `LERC_EXPORTS` so it still doesn't build correctly for mingw... I don't know what's the best way of handling this apart from adding the definition manually in CmakeLists.txt, or changing the project name to uppercase...

Btw, did you add `LERC_EXPORTS` manually to the Visual Studio project file? It is supposed automatically predefine this macro so I'm surprised to see it in the project file explicitly... I don't have VS to check this, but can you see if there is already a predefined macro under "Properties -> C/C++ -> Preprocessor -> Preprocessor Definition" and if it is all uppercase?

File: src/LercLib/include/Lerc_c_api.h [hunk 1]
```diff
@@ -45,15 +45,18 @@ extern "C" {
 #define LERC_AT_LEAST_VERSION(maj,min,patch) \
     (LERC_VERSION_NUMBER >= LERC_COMPUTE_VERSION(maj,min,patch))
 
-#if defined(_MSC_VER)
-  #define LERCDLL_API __declspec(dllexport)
+#ifdef _WIN32
```
Comment (kmilos) [inline src/LercLib/include/Lerc_c_api.h:48]: Might want to change to

#if defined _WIN32 || defined __CYGWIN__

to cover all Windows possibilities once and for all

**Comment by kmilos** [comment]
I'd also like to suggest this modification finally:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index fa691ff..0bfc759 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,20 +8,19 @@ set(CMAKE_CXX_STANDARD 17)
 file(GLOB SOURCES
     "src/LercLib/*"
     "src/LercLib/Lerc1Decode/*"
-    "src/LercLib/include"
+    "src/LercLib/include/*"
 )
-message(${SOURCES})
 
 add_library(Lerc SHARED ${SOURCES})
 
-target_link_libraries (Lerc)
+set_target_properties(Lerc PROPERTIES DEFINE_SYMBOL LERC_EXPORTS)
 
 install(
     TARGETS Lerc
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    PUBLIC_HEADER DESTINATION  ${CMAKE_INSTALL_INCLUDEDIR}
+    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
 
 install(
```

**Comment by tmaurer3** [comment]
@kmilos  I had to add LERC_EXPORTS to the VS project settings, there was nothing there.

**Comment by kmilos** [comment]
Great, thanks!

**Pull request merged**